### PR TITLE
Implements SingleAttestation handling

### PR DESCRIPTION
**Pull request by tbenr**
Changes related to API are are not part of this PR.
I added non-related debug logs in `ThrottlingSyncSource` (I think we need them)

- implements VC side, where we dynamically select which attestation type to produce
- implements BN side, where we validate it and convert it to aggregated attestation during processing
- `ValidatableAttestation::getAttestation` will return the converted attestation
- `ValidatableAttestation::getUnconvertedAttestation` will return the original single attestation, mainly used for EventSubscription

related to https://github.com/Consensys/teku/issues/8804

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/util/AttestationUtilElectra.java [hunk 1]
```diff
@@ -93,4 +115,162 @@ public AttestationData getGenericAttestationData(
...
+        singleAttestationAggregationBits,
+        singleAttestation.getData(),
+        singleAttestation.getAggregateSignature(),
+        attestationElectraSchema
+            .getCommitteeBitsSchema()
+            .orElseThrow()
+            .ofBits(singleAttestation.getFirstCommitteeIndex().intValue()));
+  }
+
+  private SafeFuture<AttestationProcessingResult> validateSingleAttestationSignature(
```
Comment (mehdi-aouadi) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/util/AttestationUtilElectra.java:213]: I guess we could refactor `AttestationUtil::isValidIndexedAttestationAsync` and reuse the signature verification

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/util/AttestationUtilElectra.java [hunk 2]
```diff
@@ -93,4 +115,162 @@ public AttestationData getGenericAttestationData(
...
+      final BeaconState state, final SingleAttestation singleAttestation) {
+    final IntList committee =
+        beaconStateAccessors.getBeaconCommittee(
+            state,
+            singleAttestation.getData().getSlot(),
+            singleAttestation.getFirstCommitteeIndex());
+
+    int validatorIndex = singleAttestation.getValidatorIndexRequired().intValue();
+
+    int validatorCommitteeBit = committee.indexOf(validatorIndex);
```
Comment (mehdi-aouadi) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/util/AttestationUtilElectra.java]: These two can be final

File: networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/subnets/AttestationSubnetSubscriptionsTest.java [hunk 3]
```diff
@@ -148,6 +148,24 @@ void shouldUnsubscribeFromOnlyCommitteeOnSubnet() {
...
+        new AttestationSubnetSubscriptions(
+            spec,
+            asyncRunner,
+            gossipNetwork,
+            gossipEncoding,
+            storageSystem.recentChainData(),
+            processor,
+            storageSystem.recentChainData().getCurrentForkInfo().orElseThrow(),
+            DebugDataDumper.NOOP);
+    subnetSubscriptions.getAttestationSchema().toSingleAttestationSchemaRequired();
```
Comment (mehdi-aouadi) [inline networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/subnets/AttestationSubnetSubscriptionsTest.java]: I'd suggest adding an assertion here, maybe `assertDoesNotThrow`

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/attestations/AttestationProductionDuty.java [hunk 4]
```diff
@@ -136,19 +138,44 @@ private List<SafeFuture<ProductionResult<Attestation>>> produceAttestationsForCo
...
+  private SignedAttestationProducer selectSignedAttestationProducer(final UInt64 slot) {
+    final SchemaDefinitions schemaDefinitions = spec.atSlot(slot).getSchemaDefinitions();
+
+    return schemaDefinitions
+        .toVersionElectra()
+        .<SignedAttestationProducer>map(
+            schemaDefinitionsElectra ->
+                (a, b, c) ->
+                    createSignedSingleAttestation(
+                        schemaDefinitionsElectra.getSingleAttestationSchema(), a, b, c))
```
Comment (mehdi-aouadi) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/attestations/AttestationProductionDuty.java]: I'd rename the `a, b, c` to something more relevant: `attestationData, validator, signature`

**Review by mehdi-aouadi — COMMENTED** [review]
Overall the logic and flow LGTM, just some nits and possible refactoring

**Comment by tbenr** [comment]
> Overall the logic and flow LGTM, just some nits and possible refactoring

Thanks for the good feedback! I should have addressed all of them

**Review by mehdi-aouadi — APPROVED** [review]
LGTM
Nice clean up and refactoring!

**Review by rolfyone — APPROVED** [review]
found it super interesting that we clearly see aggregations vs. attestations now on the event stream, that could be fun... looks good.

**Pull request merged**